### PR TITLE
Add Gradient-Free Fit Method to CLV Models

### LINKED ISSUE #1224: `ParetoNBDModel` Maintenance Tasks
- [x] Suppress `pytensor` Hyp2F1 Opt warning
- [x] Add `DEMetropolisZ` fitting option
- [x] Fix `thin_fit_result` (`DEMetropolisZ` support may fix this)

### PR DESCRIPTION
**Pull request by ColtAllen**
## Description

This PR adds the gradient-free `pm.DEMetropolisZ` sampler as a fit option. It is primarily for `ParetoNBDModel`, but it made more sense to add it to the `CLVBaseModel` class.

I've also removed the `arviz` plots from the Pareto/NBD notebook as there is a new warning involving discrete sampling and they are not displaying properly. There is an open issue on adding them as a plotting function.

## Related Issue

- [ ] Closes #1224
- [ ] Related to  https://github.com/pymc-labs/pymc-marketing/issues/1226

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [x] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [ ] MMM
- [x] CLV

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [x] Documentation
- [x] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--1266.org.readthedocs.build/en/1266/

### CODE REVIEW

File: docs/source/notebooks/mmm/dag [hunk 1]
```diff
@@ -0,0 +1,10 @@
+digraph {
+	graph [dpi=200]
+	media [label="media spend"]
+	seas [label=seasonality]
+	trend [label=trend]
+	y [label=sales]
+	media -> y
+	seas -> y
+	trend -> y
+}
```
Comment (juanitorduz) [inline docs/source/notebooks/mmm/dag:10]: Why this change?

**Review by juanitorduz — APPROVED** [review]
Great! Thanks @ColtAllen !

File: docs/source/notebooks/mmm/dag (hunk 1 above)
Comment (ColtAllen) [reply docs/source/notebooks/mmm/dag:10]: It wasn't intentional - I was updating my dev dependencies via the Makefile and accidentally ran the UML hook and notebook runner. Want me to remove these files?

Comment (juanitorduz) [reply docs/source/notebooks/mmm/dag:10]: No, it think it should be fine :)
